Assets/Vimeo/Scripts/Player/VimeoPlayer.cs

```diff
@@ -87,41 +71,86 @@ public override void SignIn(string _token)
             if (vimeo_url != null && vimeo_url != "") {
                 vimeoVideo = null;
                 string[] matches = Regex.Split(vimeo_url, "(vimeo.com)?(/channels/[^/]+)?/?([0-9]+)"); // See https://regexr.com/3prh6
                 if (matches[3] != null) {
                     vimeoVideoId = matches[3];
-                    LoadVimeoVideoById(int.Parse(vimeoVideoId));
+                    LoadVideo(int.Parse(vimeoVideoId));
                 }
                 else {
-                    Debug.LogWarning("Invalid Vimeo URL");
+                    Debug.LogError("[Vimeo] Invalid Vimeo URL");
                 }
             }
         }
 
-        public void LoadVimeoVideoById(int vimeo_id)
+        public void LoadVideo(int vimeo_id)
         {
-            if (!vimeoSignIn) {
-                Debug.LogWarning("You have not signed into the Vimeo Player.");
-            }
+            loadingVideoMetadata = true;
+            videoControllerReady = false;
 
-            controller.videoScreenObject = videoScreen;
-            controller.Setup();
+            SetupVideoController();
 
             if (videoScreen == null && videoPlayerType == VideoPlayerType.UnityPlayer) {
-                Debug.LogWarning("No video screen was specified.");
+                Debug.LogWarning("[Vimeo] No video screen was specified.");
             }
 
-            api.OnRequestComplete += OnLoadVimeoVideoComplete;
+            api.OnRequestComplete += VideoMetadataLoad;
             api.GetVideoFileUrlByVimeoId(vimeo_id);
         }
 
+        private void SetupVideoController()
+        {
+            // TODO abstract this out into a VideoPlayerManager (like EncoderManager.cs)
+            if (videoPlayerType == VideoPlayerType.UnityPlayer) {
+                if (controller == null) {
+                    controller = gameObject.AddComponent<VideoController>();
+                    controller.playerSettings = this;
+                    controller.videoScreenObject = videoScreen;
+
+                    controller.OnVideoStart += VideoStarted;
+                    controller.OnPlay       += VideoPlay;
+                    controller.OnPause      += VideoPaused;
+                    controller.OnFrameReady += VideoFrameReady;
+                            
+                    if (audioSource && audioSource is AudioSource) {
+                        if (audioSource != null) {
+                            controller.audioSource = audioSource;
+                        }
+                        else {
+                            videoScreen.gameObject.AddComponent<AudioSource>();
+                        }
+                    }
+                }
+                else if (videoPlayerType == VideoPlayerType.UnityPlayer) {
+                    controller.videoScreenObject = videoScreen;
+                    controller.Setup();
+                }
+            }
+            else if (videoPlayerType == VideoPlayerType.AVProVideo && mediaPlayer == null) {
```

Comment:
I get an error from this line
`Assets/Vimeo/Scripts/Player/VimeoPlayer.cs(132,71): error CS0103: The name mediaPlayer does not exist in the current context`